src/library.js

```diff
@@ -52,13 +52,7 @@ mergeInto(LibraryManager.library, {
   // JavaScript <-> C string interop
   // ==========================================================================
 
-  $stringToNewUTF8__deps: ['malloc'],
-  $stringToNewUTF8: function(jsString) {
-    var length = lengthBytesUTF8(jsString)+1;
-    var cString = _malloc(length);
-    stringToUTF8(jsString, cString, length);
-    return cString;
-  },
+  $stringToNewUTF8: '$allocateUTF8',
```

Comment:
This creates an alias? And we can't remove it entirely because it might be used by user code? Is there a way to mark it deprecated?

Reply:
Yes, this creates an alias.

There is internal usage of `stringToNewUTF8`.   Perhaps as a followup we can remove all the internal usage and move this alias into `src/library_legacy.js`.